### PR TITLE
Bug 1824717 - Re-evaluate usage of robolectric in AC unit tests

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1824717

### CODE REVIEW

**Comment by mcarare** [comment]
Test times (M1 MacBook)

**With RobolectricTestRunner removed**

|   Test	|   Before	|   After	|
|---	|---	|---	|
|  TabsUseCasesKtTest 	|  3 s 228 ms 	|   1 s 524 ms	|

File: android-components/components/feature/readerview/src/main/assets/extensions/readerview/readability/readability-0.4.2.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ Readability.prototype = {
   DEPRECATED_SIZE_ATTRIBUTE_ELEMS: [ "TABLE", "TH", "TD", "HR", "PRE" ],
 
   // The commented out elements qualify as phrasing content but tend to be
-  // removed by readability when put into paragraphs, so we ignore them here.
+   by readability when put into paragraphs, so we ignore them here.
```
Comment (jonalmeida) [inline android-components/components/feature/readerview/src/main/assets/extensions/readerview/readability/readability-0.4.2.js:158]: oops

**Comment by mcarare** [comment]
Locally, the duration of running **all tests** in AC decreased from `8 m 19 s 66` ms to `7 m 43 s 165 ms`.
The total time from Gradle ( if tests didn't run in parallel) decreased from `55 m 15 s ` to `52 m 17 s`.

**Review by jonalmeida — APPROVED** [review]
Doesn't look like anything has changed since I last reviewed this patch. LGTM! 👍 

Do you think there is any value in re-running CI to check for intermittent failures? We have a week of bake time until the nightly cut-off so the sooner this lands, the better.

**Pull request merged**